sql/core/src/test/scala/org/apache/spark/sql/DateFunctionsSuite.scala

```diff
@@ -545,6 +545,50 @@ class DateFunctionsSuite extends QueryTest with SharedSparkSession {
       df1.select(to_date(col("x"))), Row(Date.valueOf("2016-02-29")) :: Row(null) :: Nil)
   }
 
+  test("function unix_date") {
+    val d1 = Date.valueOf("2015-07-22")
+    val d2 = Date.valueOf("2015-07-01")
+    val d3 = Date.valueOf("2014-12-31")
+    val df = Seq(d1, d2, d3).toDF("d")
+
+    checkAnswer(
+      df.select(unix_date(col("d"))),
+      Seq(Row(16435), Row(16617), Row(16638)))
```

Comment:
can we compare the result with `df.selectExpr("unix_date(d)")`? the numbers here are hard to understand